### PR TITLE
Updated the code, state_size was pointing to the wrong value

### PR DESCRIPTION
**Pull request by hpandeycodeit**
JIRA: **MADLIB-1325**

`state_size` was not updating and always has the wrong value ( -1)
Updated the code to point it to the correct value. 

**Description** : 
As per MADLIB-1325, in  `mlp_classification` when a tolerance value if provided and the Loss generated reaches the below the tolerance value, the iteration should stop. But as of now, it doesn't consider the tolerance value and keep running to all the iterations. 

This PR fixes the above issue. As the `state_size` not points to the correct array index.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/934/

File: src/ports/postgres/modules/convex/mlp_igd.py_in [hunk 1]
```diff
@@ -334,7 +334,7 @@ def mlp(schema_madlib, source_table, output_table, independent_varname,
                         """
                 it.update(train_sql)
                 if it_args['state_size'] == -1:
-                    it_args['state_size'] = it.get_state_size()
+                    it.kwargs['state_size'] = it.get_state_size()
```
Comment (njayaram2) [inline src/ports/postgres/modules/convex/mlp_igd.py_in]: This if condition will be True for every iteration since `it_args['state_size']` is not updated after the first time it goes into the if statement. So we will end up running the query in `it.get_state_size()` multiple times which is unnecessary (the `state_size` will not change). Can we update `it_args['state_size']` too inside this if statement (or something similar which will preclude this if statement from evaluating to True after the first time)? Something simple would be, but feel free to do anything else that might be cleaner/efficient:
```
if it_args['state_size'] == -1:
    it.kwargs['state_size'] = it.get_state_size()
    it_args['state_size'] = it.kwargs['state_size']
```

Comment (hpandeycodeit) [reply src/ports/postgres/modules/convex/mlp_igd.py_in]: Thanks for pointing it out! Yeah, I missed correcting the `if condition` here. Fixed it now.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/madlib-pr-build/941/

**Comment by fmcquillan99** [comment]
No tolerance :

```
madlib=# SELECT madlib.mlp_classification(
madlib(#     'iris_data',      -- Source table
madlib(#     'mlp_model',      -- Destination table
madlib(#     'attributes',     -- Input features
madlib(#     'class_text',     -- Label
madlib(#     ARRAY[5],         -- Number of units per layer
madlib(#     'learning_rate_init=0.003,
madlib'#     n_iterations=25,
madlib'#     tolerance=0',     -- Optimizer params
madlib(#     'tanh',           -- Activation function
madlib(#     NULL,             -- Default weight (1)
madlib(#     FALSE,            -- No warm start
madlib(#     TRUE             -- Not verbose
madlib(# );
INFO:  Iteration: 1, Loss: <1.57403083382>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 2, Loss: <1.18209701678>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 3, Loss: <0.805838101786>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 4, Loss: <0.456743716108>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 5, Loss: <0.269013324379>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 6, Loss: <0.176825519276>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 7, Loss: <0.12831471301>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 8, Loss: <0.0995580174594>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 9, Loss: <0.0808205589525>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 10, Loss: <0.0679497236685>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 11, Loss: <0.058451236734>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 12, Loss: <0.0510716510938>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 13, Loss: <0.0453356242116>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 14, Loss: <0.040657636203>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 15, Loss: <0.0368149574231>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 16, Loss: <0.0336085514468>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 17, Loss: <0.0309155266308>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 18, Loss: <0.0286056560803>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 19, Loss: <0.0266028934429>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 20, Loss: <0.0248412801724>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 21, Loss: <0.023296777543>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 22, Loss: <0.0219201410755>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 23, Loss: <0.0206909609214>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 24, Loss: <0.0195874516687>
CONTEXT:  PL/Python function "mlp_classification"
 mlp_classification 
--------------------
 
(1 row)
```


tolerance=0.1 :

```
madlib=# SELECT madlib.mlp_classification(
madlib(#     'iris_data',      -- Source table
madlib(#     'mlp_model',      -- Destination table
madlib(#     'attributes',     -- Input features
madlib(#     'class_text',     -- Label
madlib(#     ARRAY[5],         -- Number of units per layer
madlib(#     'learning_rate_init=0.003,
madlib'#     n_iterations=25,
madlib'#     tolerance=0.1',     -- Optimizer params
madlib(#     'tanh',           -- Activation function
madlib(#     NULL,             -- Default weight (1)
madlib(#     FALSE,            -- No warm start
madlib(#     TRUE             -- Not verbose
madlib(# );
INFO:  Iteration: 1, Loss: <1.47220611074>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 2, Loss: <1.24578776394>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 3, Loss: <0.887573516227>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 4, Loss: <0.530981965126>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 5, Loss: <0.306411380721>
CONTEXT:  PL/Python function "mlp_classification"
INFO:  Iteration: 6, Loss: <0.195035410704>
CONTEXT:  PL/Python function "mlp_classification"
 mlp_classification 
--------------------
 
(1 row)
```

LGTM

@hpandeycodeit @njayaram2   Just to confirm:  in the case of grouping, all groups must meet the tolerance threshold before training is ended?
